tests/ui/default_numeric_fallback.rs

```diff
@@ -1,99 +1,18 @@
 #![warn(clippy::default_numeric_fallback)]
 #![allow(unused)]
-#![allow(clippy::never_loop)]
-#![allow(clippy::no_effect)]
-#![allow(clippy::unnecessary_operation)]
-
-fn ret_i31() -> i32 {
-    23
-}
```

Comment:
I think you should keep all the tests, so if someone else wants to enhance/modify the lint there are already tests in place, that can be used for reference. Also then I will see, how my suggested changes changed the behavior of the lint.